modules/tensor_mechanics/src/materials/ComputeEigenstrainBeamFromVariable.C

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeEigenstrainBeamFromVariable.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeEigenstrainBeamFromVariable);
+
+template <>
+InputParameters
+validParams<ComputeEigenstrainBeamFromVariable>()
+{
+  InputParameters params = validParams<ComputeEigenstrainBeamBase>();
+  params.addClassDescription("Computes an eigenstrain from a set of variables");
+  params.addCoupledVar("disp_eigenstrain",
```

Comment:
I'd rename these to `displacement_eigenstrain_variables` and `rotational_eigenstrain_variables`.